modin/pandas/dataframe.py

```diff
@@ -448,11 +448,7 @@ def groupby(
         elif hashable(by) and not isinstance(by, pandas.Grouper):
             drop = by in self.columns
             idx_name = by
-            if (
-                self._query_compiler.has_multiindex(axis=axis)
-                and by in self._query_compiler.get_index_names(axis)
-                and by is not None
-            ):
+            if by in self._query_compiler.get_index_names(axis) and by is not None:
```

Comment:
```suggestion
            if by is not None and by in self._query_compiler.get_index_names(axis):
```
looks more natural in this way

Reply:
fixed